### DISCUSSION TITLE
Handling with multiple providers

### DISCUSSION THREAD

**Discussion by zcharlesallen** [polymind-inc/acmebot#621] [Q&A] [answered]
I have a bit of a unique situation.  Currently, we register our domains through GoDaddy, and most of those domains have their DNS provided through it as well; however, we have some other domains with their DNS provided through AWS.  I'm not sure on how Acmebot is handling a single domain that exists in both entities.  Is it possible to direct it so that DNS entries are entered in the appropriate entity?  My attempts at creating certificates where the domain is registered in GoDaddy but DNS is hosted through AWS.

**Comment by shibayan** [comment, marked as answer]
Currently, DNS zones obtained from each DNS Provider are processed together, so if each DNS Provider has a DNS zone with the same name, this will cause unintended behavior.

This behavior is also the result of a compromise to avoid certificate issuance failures when transitioning between DNS Providers, e.g., changing DNS zones from GoDaddy to AWS.

Conversely, if the issues involved in migrating DNS zones can be resolved, it can be handled. For example, if a DNS zone managed by GoDaddy is lost, we can check to see if it is registered with another DNS provider and migrate it automatically.

Edited:

- https://github.com/shibayan/keyvault-acmebot/pull/623

**Comment by zcharlesallen** [comment]
Thank you for the response and this make sense.  Helps me determine what I need to do.
